src/react-redux.js

```diff
@@ -71,7 +71,7 @@ export function connect<SP, P, C: ComponentType<P>>(
   mapStateToProps?: (GlobalState, OwnProps<C,
     // Error "property `foo` is missing"?  Add to inner component's props.
     SP>) => SP,
-): C => ComponentType<OwnProps<C, SP>> {
+): C => ComponentType<$ReadOnly<OwnProps<C, SP>>> {
```

Comment:
> [...] which we know because you've prepared those follow-on commits and they pass tests.

Well, that and the fact that React requires props to be immutable anyway. If this commit resulted in any declared types being tightened beyond their existing use, that would reveal an actual bug in the code.

(Clarified accordingly, though.)

Reply:
Mmm, true enough!